File: ChainOfResponsibility/PHP/FilterChain.php
<?php
  class FilterChain implements Filter {
    private $filters;

    public function __construct() {
      $this->filters = array();
    }

    public function addFilter($filter) {
      array_push($this->filters, $filter);
      return $this;
    }

    public function handleFilter($str) {
      foreach ($this->filters as $filter) {
        $str = $filter->handleFilter($str);
      }

      return $str;
    }
  }

File: ChainOfResponsibility/Go/chain/FilterChain.go
package chain

type FilterChain struct {
  filters []Filter
}

func NewFilterChain() *FilterChain {
  return &FilterChain{}
}

func (this *FilterChain) AddFilter(filter Filter) *FilterChain {
  this.filters = append(this.filters, filter)
  return this
}

func (this *FilterChain) HandleFilter(str string) string {
  for _, filter := range this.filters {
    str = filter.HandleFilter(str)
  }

  return str
}
